connector/spanmetricsconnector/connector.go

```diff
@@ -391,48 +391,26 @@ func (p *connectorImp) aggregateMetrics(traces ptrace.Traces) {
 				}
 
 				key := p.buildKey(serviceName, span, p.dimensions, resourceAttr)
-				var attributesFun metrics.BuildAttributesFun
-
-				// Note: we check cardinality limit here for sums metrics but it is the same
-				// for histograms because both use the same key and attributes.
-				if rm.sums.IsCardinalityLimitReached() {
-					attributesFun = func() pcommon.Map {
-						attributes := pcommon.NewMap()
-						for _, d := range p.dimensions {
-							if v, exists := utilattri.GetDimensionValue(d, span.Attributes(), resourceAttr); exists {
-								v.CopyTo(attributes.PutEmpty(d.Name))
-							}
-						}
-						attributes.PutBool(overflowKey, true)
-
-						return attributes
-					}
-				} else {
-					attributesFun = func() pcommon.Map {
-						attributes := p.buildAttributes(
-							serviceName,
-							span,
-							resourceAttr,
-							p.dimensions,
-							ils.Scope(),
-						)
-						return attributes
-					}
+				attributesFun := func() pcommon.Map {
+					return p.buildAttributes(serviceName, span, resourceAttr, p.dimensions, ils.Scope())
 				}
 
-				if !p.config.Histogram.Disable {
-					// aggregate histogram metrics
-					h := histograms.GetOrCreate(key, attributesFun, startTimestamp)
-					p.addExemplar(span, duration, h)
-					h.Observe(duration)
-				}
 				// aggregate sums metrics
-				s := sums.GetOrCreate(key, attributesFun, startTimestamp)
-				if p.config.Exemplars.Enabled && !span.TraceID().IsEmpty() {
+				s, limitReached := sums.GetOrCreate(key, attributesFun, startTimestamp)
+				if !limitReached && p.config.Exemplars.Enabled && !span.TraceID().IsEmpty() {
 					s.AddExemplar(span.TraceID(), span.SpanID(), duration)
 				}
 				s.Add(1)
 
+				// aggregate histogram metrics
+				if !p.config.Histogram.Disable {
```

Comment:
Should we also add tests for the histograms? I didn't add them before because I though they were not needed but I'm not sure now.

Reply:
yes, I have also add a test case for Histogram in `TestConnectorWithCardinalityLimit`